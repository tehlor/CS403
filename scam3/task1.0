(println (scoping 'cs403sucks ((define (zirp x) this) 5)))
