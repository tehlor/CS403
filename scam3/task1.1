(println (scoping '* ((define (zirp x) this) 5)))
