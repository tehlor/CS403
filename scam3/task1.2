(println (scoping 'x ((define (zirp x) this) 5)))
