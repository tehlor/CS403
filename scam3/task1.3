(inspect (scoping 'x this))
(inspect (scoping 'this this))
(inspect (scoping 'quote this))
