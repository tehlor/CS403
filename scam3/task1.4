(define (zorp a) (lambda (b) this))
(define obj ((zorp 5) 8))
(inspect (scoping 'x obj))
(inspect (scoping 'a obj))
(inspect (scoping 'b obj))
(inspect (scoping 'define obj))
