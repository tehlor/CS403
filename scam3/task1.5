(define (zorp a) (lambda (b) (lambda (c) (define d 3) this)))
(define obj (((zorp 5) 8) 'x))
(inspect (scoping 'x obj))
(inspect (scoping 'a obj))
(inspect (scoping 'b obj))
(inspect (scoping 'c obj))
(inspect (scoping 'd obj))
(inspect (scoping 'lambda obj))