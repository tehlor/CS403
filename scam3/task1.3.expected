(scoping (quote x) this) is undefined
(scoping (quote this) this) is bound
(scoping (quote quote) this) is free
